hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/SendContainerRequestHandler.java

```diff
@@ -80,6 +82,14 @@ public void onNext(SendContainerRequest req) {
         LOG.info("Accepting container {}", req.getContainerID());
       }
 
+      // check and avoid download of container file if target already have
+      // container data and import in progress
+      if (!importer.isAllowedContainerImport(containerId)) {
+        throw new StorageContainerException("Container exists or " +
+            "import in progress with container Id " + containerId,
+            ContainerProtos.Result.CONTAINER_EXISTS);
+      }
```

Comment:
We can move this check before the `if` block (and change it to use `req.getContainerID()`) to avoid creating some resources unnecessarily.

Reply:
@adoroszlai all comments updated